pkg/api/controller_test.go

```diff
@@ -216,6 +245,189 @@ func TestController_CommitsGetBranchCommitLogHandler(t *testing.T) {
 	})
 }
 
+func TestController_CommitsGetBranchCommitLogByPath(t *testing.T) {
+	clt, deps := setupClientWithAdmin(t, "")
+	ctx := context.Background()
+	/*
+			This is the tree we test on:
+
+		                      P              branch "branch-b"
+		                     / \
+		                   	/   \
+		 .---A---B---C-----D-----R---N---X-  branch "main"
+		              \             /
+		               \           /
+		                ---L---M---          branch "branch-a"
+
+				Commits content:
+				•	commit A
+					⁃	added data/a/a.txt
+					⁃	added data/a/b.txt
+					⁃	added data/a/c.txt
+					⁃	added data/b/b.txt
+				•	commit B
+					⁃	added data/a/foo.txt
+					⁃	added data/b/bar.txt
+				•	commit C
+					⁃	changed data/a/a.txt
+				•	commit D
+					⁃	changed data/b/b.txt
+				•	commit P
+					⁃	added data/c/banana.txt
+				•	commit R
+					⁃	merged branch-a to main
```

Comment:
Did you mean to write commit R - merged branch-b to main and commit N merged branch-a to main?

Reply:
Yes, thanks!